query-node/mappings/src/derivedPropertiesManager/applications/videoNftCollectors.ts

```diff
@@ -0,0 +1,145 @@
+import { DerivedPropertiesManager } from '../classes'
+import { IExecutor, IListener, IChangePair } from '../interfaces'
+import { DatabaseManager } from '@joystream/hydra-common'
+import { Channel, ChannelNftCollectors, CuratorGroup, Membership, OwnedNft } from 'query-node/dist/model'
+
+export type IOwnedNftChange = 1 | -1
+
+class OwnedNftUpdateListener implements IListener<OwnedNft, IOwnedNftChange> {
+  getRelationDependencies(): string[] {
+    return ['ownerMember']
+  }
+
+  hasValueChanged(oldValue: OwnedNft | undefined, newValue: OwnedNft): IChangePair<IOwnedNftChange> | undefined
+  hasValueChanged(oldValue: OwnedNft, newValue: OwnedNft | undefined): IChangePair<IOwnedNftChange> | undefined
+  hasValueChanged(oldValue: OwnedNft, newValue: OwnedNft): IChangePair<IOwnedNftChange> | undefined {
+    // at least one video should always exists but due to TS type limitations
+    // (can't define at least one-of-two parameters required) this safety condition needs to be here
+    if (!oldValue && !newValue) {
+      return undefined
+    }
+
+    // NFT is being created?
+    if (!oldValue) {
+      return {
+        old: undefined,
+        new: 1,
+      }
+    }
+
+    // NFT is being deleted?
+    if (!newValue) {
+      return {
+        old: -1,
+        new: undefined,
+      }
+    }
+
+    // owner is unchanged?
+    if (oldValue.ownerMember?.id.toString() === newValue.ownerMember?.id.toString()) {
+      return undefined
+    }
+
+    // owner changed
+    return {
+      old: -1,
+      new: 1,
+    }
+  }
+}
+
+interface IOwnerType {
+  member: Membership | undefined
+  curatorGroup: CuratorGroup | undefined
+}
+
+class NftCollectorsExecutor implements IExecutor<OwnedNft, IOwnedNftChange, ChannelNftCollectors> {
+  async loadDerivedEntities(store: DatabaseManager, ownedNft: OwnedNft): Promise<ChannelNftCollectors[]> {
+    // TODO: find way to reliably decide if channel, etc. are loaded and throw error if not
```

Comment:
Do we want to resolve this before merging?

Reply:
No, I will fix this as a separate issue later on (either with an active video counter fix or later when implementing forum post counters).